src/main/java/ru/hogwarts/school/controller/StudentController.java

```diff
@@ -0,0 +1,63 @@
+package ru.hogwarts.school.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.service.StudentService;
+
+import java.util.Collections;
+import java.util.List;
+
+@RestController
+@RequestMapping("student")
+public class StudentController {
+    private final StudentService studentService;
+    private int age;
+
+    public StudentController(StudentService studentService) {
+        this.studentService = studentService;
+    }
+
+    @PostMapping
+    public Student createStudent(@RequestBody Student student) {
+        return studentService.createStudent(student);
+    }
+
+    @GetMapping("{studentId}")
+    public ResponseEntity<Student> getStudent(@PathVariable Long studentId) {
+        Student student = studentService.getStudentById(studentId);
+        if (student == null) {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok(student);
+    }
+
+    @DeleteMapping("{studentId}")
+    public ResponseEntity<Void> deleteStudent(@PathVariable Long studentId) {
+        studentService.deleteStudent(studentId);
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping("getFaculty/{studentId}")
```

Comment:
неправильный URI с точки зрения REST. Идёт обращение по id к ресурсу, затем уже к его характеристике. Поэтому должно быть так

```suggestion
    @GetMapping("/{studentId}/faculty")
```